OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -629,6 +630,165 @@ public static DreamValue NativeProc_clamp(DreamObject instance, DreamObject usr,
+
+            List<DreamValue> GradientList;
+
+            if (arguments.GetArgument(0, "A").TryGetValueAsDreamList(out DreamList gradlist)) {
+                GradientList = gradlist.GetValues();
+
+                if (gradlist.ContainsValue(new DreamValue("loop"))) { loop = true; }
+
+                DreamValue dictspace = gradlist.GetValue(new("space"));
+                if (dictspace != DreamValue.Null && dictspace.TryGetValueAsInteger(out int dictlookup)) { colorspace = dictlookup; }
+
+            } else {
+                GradientList = arguments.GetAllArguments();
+            }
+
+            if (arguments.OrderedArguments.Contains(new("loop"))) { loop = true; }
+
+            if (arguments.NamedArguments.TryGetValue("space", out DreamValue namedlookup) && namedlookup.TryGetValueAsInteger(out int space)) { colorspace = space; }
+
+            /// true: look for int: false look for color
+            bool color_or_int = true;
+
+            float workingfloat = 0;
+            float maxvalue = 1;
+            float minvalue = 0;
+            float leftboundy = 0;
+            float rightboundry = 1;
+
+            /// None of these should be null however C# is "special"
+            Color left = new();
+            Color right = new();
+
+            bool leftexists = false;
+            bool rightexists = false;
+
+            foreach (DreamValue value in GradientList) {
+                if (color_or_int && value.TryGetValueAsFloat(out float flt)) { // Int
+                    color_or_int = false;
+                    workingfloat = flt;
+                    maxvalue = Math.Max(maxvalue, flt);
+                    minvalue = Math.Min(minvalue, flt);
+                    continue; // Succesful parse
+                }
+
+                if (value.TryGetValueAsString(out string? strvalue) && ColorHelpers.TryParseColor(strvalue, out Color color)) {
+
+                    if(loop && index >= maxvalue) {
+                        index %= maxvalue;
+                    }
+
+                    if (workingfloat >= index) {
+                        right = color;
+                        rightboundry = workingfloat;
+                        rightexists = true;
+                        break;
+                    }
+                    else {
+                        left = color;
+                        leftexists = true;
+                        leftboundy = workingfloat;
+                    }
+
+                    if (color_or_int) {
+                        workingfloat = 1;
+                    }
+
+                    color_or_int = true;
+                }
+            }
+
+            /// Convert the index to a 0-1 range
+            float normalized = (index - leftboundy) / (rightboundry - leftboundy);
+
+            /// Cheap way to make sure the gradient works at the extremes (eg 1 and 0)
+            if (!leftexists || (rightexists && normalized == 1) || (rightexists && normalized == 0)) {
+                if (right.A == 1) {
```

Comment:
Use `AByte` to avoid potential float issues.
```suggestion
                if (right.AByte == 255) {
```